plugin/src/middleware/response.ts

```diff
@@ -0,0 +1,95 @@
+import { NextResponse } from 'next/server'
+
+import type { ElementHandlers } from './html-rewriter'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type NextDataTransform = <T extends { pageProps?: Record<string, any> }>(props: T) => T
```

Comment:
[dust] Prefer `unknown`

```suggestion
export type NextDataTransform = <T extends { pageProps?: Record<string, unknown> }>(props: T) => T
```

Reply:
The reason I didn't do this is because it would mean that users would need to create a type and explicitly cast the props value whenever they used `transformData`, which could be quite cumbersome. I'll add a comment.